schema-engine/sql-schema-describer/src/postgres.rs

```diff
@@ -1531,6 +1435,144 @@ impl<'a> SqlSchemaDescriber<'a> {
+        let table_name = row.get_expect_string("table_name");
+        let table_id: TableId = match table_ids.get(&(namespace, table_name)) {
+            Some(id) => *id,
+            None => continue,
+        };
+
+        let column_name = row.get_expect_string("column_name");
+
+        let column_id = match sql_schema.walk(table_id).column(&column_name) {
+            Some(col) => col.id,
+            _ => continue,
+        };
+
+        let index_name = row.get_expect_string("index_name");
+        let is_unique = row.get_expect_bool("is_unique");
+        let is_primary_key = row.get_expect_bool("is_primary_key");
+        let column_index = row.get_expect_i64("column_index");
+        let index_algo = row.get_expect_string("index_algo");
+
+        let sort_order = row.get_string("column_order").map(|v| match v.as_ref() {
+            "ASC" => SQLSortOrder::Asc,
+            "DESC" => SQLSortOrder::Desc,
+            misc => panic!("Unexpected sort order `{misc}`, collation should be ASC, DESC or Null"),
+        });
+
+        let algorithm = if circumstances.contains(Circumstances::Cockroach) {
+            match index_algo.as_str() {
+                "inverted" => SqlIndexAlgorithm::Gin,
+                _ => SqlIndexAlgorithm::BTree,
+            }
+        } else {
+            match index_algo.as_str() {
+                "btree" => SqlIndexAlgorithm::BTree,
+                "hash" => SqlIndexAlgorithm::Hash,
+                "gist" => SqlIndexAlgorithm::Gist,
+                "gin" => SqlIndexAlgorithm::Gin,
+                "spgist" => SqlIndexAlgorithm::SpGist,
+                "brin" => SqlIndexAlgorithm::Brin,
+                other => {
+                    tracing::warn!("Unknown index algorithm on {index_name}: {other}");
+                    SqlIndexAlgorithm::BTree
+                }
+            }
+        };
+
+        if column_index == 0 {
+            // new index!
+            let index_id = if is_primary_key {
+                sql_schema.push_primary_key(table_id, index_name)
+            } else if is_unique {
+                sql_schema.push_unique_constraint(table_id, index_name)
+            } else {
+                sql_schema.push_index(table_id, index_name)
+            };
+
+            if is_primary_key || is_unique {
+                let mut constraint_options = BitFlags::empty();
+
+                if let Some(true) = row.get_bool("condeferrable") {
+                    constraint_options |= ConstraintOption::Deferrable;
+                }
+
+                if let Some(true) = row.get_bool("condeferred") {
+                    constraint_options |= ConstraintOption::Deferred;
+                }
+
+                pg_ext
+                    .constraint_options
+                    .insert(Constraint::Index(index_id), constraint_options);
+            }
+
+            current_index = Some(index_id);
+        }
+
+        let index_id = current_index.unwrap();
```

Comment:
To Fix: When removing the filter for expression indexes in `indexes_query.sql`, this will crash